src/components/business_table.tsx

```diff
@@ -0,0 +1,195 @@
+"use client";
+
+import * as React from "react";
+import { useTransition } from "react";
+import { useRouter } from "next/navigation";
+import { Sidebar } from "~/components/sidebar";
+import {
+  ColumnDef,
+  SortingState,
+  flexRender,
+  getCoreRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { ArrowUpDown } from "lucide-react";
+import { Button } from "~/components/ui/button";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "~/components/ui/table";
+import { CampaignData } from "~/models/fund";
+import InvestmentTable from "~/components/investment_table";
+import { approveBusiness } from "~/server/action/approve_business";
+
+export default function CampaignApprovalTable({ data: initialData }: { data: CampaignData[] }) {
+  const filteredData = initialData.filter(campaign => !campaign.approve);
+  const [data, setData] = React.useState<CampaignData[]>(filteredData);
+  const [sorting, setSorting] = React.useState<SortingState>([]);
+  const [isPending, startTransition] = useTransition();
+  const router = useRouter();
+
+  const handleApprove = (rowData: CampaignData) => {
+    if (confirm(`Are you sure you want to approve the campaign for ${rowData.company}?`)) {
+      startTransition(async () => {
+        const result = await approveBusiness(Number(rowData.businessID));
+        if (result.success) {
```

Comment:
it does not reload the page on the preview deployment. I still have to refresh the page myself

Reply:
Maybe making making handler also a sync function might work